modules/cloudfront/custom_origin/custom_origin.tf

```diff
@@ -168,3 +176,19 @@ resource "aws_ssm_parameter" "cloudfront_custom_origin" {
     ]
   }
 }
+
+module "cloudfront_function" {
+  count  = var.create_cf_function == true && var.create_cf_basicauth_function == false ? 1 : 0
+  source = "../cf_function"
+
+  name      = var.cf_function_name
+  runtime   = var.cf_function_runtime
+  code_path = var.cf_function_code_path
+}
+
+module "basic_auth_cloudfront_function" {
+  count  = var.create_cf_function && var.create_cf_basicauth_function ? 1 : 0
```

Comment:
what's the difference between `create_cf_function` and `create_cf_basicauth_function` ?

Reply:
`create_cf_basicauth_function` creates the template basic auth function and `create_cf_function` is more of a "bring-your-own-function" approach, where the user could decide to bring their own CF function and apply it to the cloudfront distro. We could come up with a better naming approach